IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -568,8 +571,28 @@ Note that lookup by realmHint is a best effort (hence it is a hint), because dev
         {
             return account;
         }
-
-        if (!firstAccount) firstAccount = account;
+        
+        if (!firstAccount)
+        {
+            if (dualHeadedAccountHint)
+            {
+                if ([account.accountIdentifier.homeAccountId hasSuffix:dualHeadedAccountHint])
```

Comment:
Will OneAuth provide the account hint based on HRD, and pass the account_hint_info through SsoExt request?

Reply:
yes, that's the plan. But only for dual-headed accounts for now. There could be too much impact from this for all accounts.